### ISSUE TITLE
List of Generic Entity

### ISSUE THREAD

**Issue by wangkanai** [aspnet/EntityFramework#2014]
Its possible to code the following
```
public class GenericContext : DbContext
{
    public DbSet<Card> Cards { get; set; }
}
public class Card
{
    public int Id { get; set; }
    public List<Progress<State>> State { get; set; }
    public List<Progress<Risk>> Risk { get; set; }
}

public class Progress<T>
{
    public T State { get; set; }
    public DateTime Stamp { get; set; }
}
public enum State
{
    Active,
    Closed,
    New,
    Removed,
    Resolved
}
public enum Risk
{
    Low,
    Medium,
    High
}
```

**Comment by wangkanai** [comment]
This is issue related #1986 in anyway?

**Comment by divega** [comment]
At first glance, this should work. It should just add two new entity types to the model based on the element types on the lists.

**Comment by smitpatel** [comment]
I tried generating model like above and found a bug which is fixed via 4121fea4096609d56e2c3491154a52fe2d1c4d62
@wangkanai - You should be able to code above.
